DocumentDataAPI/DocumentDataAPI/Controllers/DocumentController.cs

```diff
@@ -0,0 +1,208 @@
+                return result == null ?
+                    NotFound() :
+                    Ok(result);
+            }
+            catch (Exception e)
+            {
+                _logger.LogError($"Unable to fetch document with id: {id}.\n{e.Message}");
+                return Problem(e.Message);
+            }
+        }
+
+        /// <summary>
+        /// Retrieves the number of documents in the database.
+        /// </summary>
+        /// <response code="200">Success: A number</response>
+        /// <response code="404">Not Found: Nothing is returned.</response>
+        /// <response code="500">Internal Server Error: a <see cref="ProblemDetails"/> describing the error.</response>
+        [HttpGet]
+        [Route("GetTotalDocumentCount")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+        public ActionResult<int> GetTotalDocumentCount()
+        {
+            try
+            {
+                int result = _repository.GetTotalDocumentCount();
+                return Ok(result);
+            }
+            catch (Exception e)
+            {
+                return Problem(e.Message);
+            }
+        }
+
+        /// <summary>
+        /// Retrieves a list of documents based on the source id.
+        /// </summary>
+        /// <response code="200">Success: A list of documents for the given source id.</response>
+        /// <response code="404">Not Found: Nothing is returned.</response>
+        /// <response code="500">Internal Server Error: a <see cref="ProblemDetails"/> describing the error.</response>
+        [HttpGet]
+        [Route("GetBySourceId/{id:int?}")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+        public ActionResult<IEnumerable<DocumentModel>> GetBySourceId(int id)
+        {
+            try
+            {
+                List<DocumentModel> result = _repository.GetBySource(id).ToList();
+                return result.Count == 0 ?
+                    NotFound() :
+                    Ok(result);
+            }
+            catch (Exception e)
+            {
+                _logger.LogError($"Unable to fetch document by source id: {id}.\n{e.Message}");
+                return Problem(e.Message);
+            }
+        }
+
+        /// <summary>
+        /// Retrieves a list of documents based on the author.
+        /// </summary>
+        /// <response code="200">Success: A list of documents by the given author.</response>
+        /// <response code="404">Not Found: Nothing is returned.</response>
+        /// <response code="500">Internal Server Error: a <see cref="ProblemDetails"/> describing the error.</response>
+        [HttpGet]
+        [Route("GetByAuthor")]
```

Comment:
Kunne det ikke lade sig gøre med query parameters? Den nuværende implementation begrænser meget ift. at man kun kan filtrere på én ting ad gangen. Med query parameters kunne man filtrere på det hele samtidigt, og DocumentSearchParameters klassen er også allerede implementeret med tilhørende funktionalitet i DocumentRepository

Reply:
Ellers er det ikke super usable 😄

Reply:
DocumentSearchParameters har indbyggede metoder til at tilføje et filter

Reply:
...Så for f.eks. et query med ?author=lars vil man så skrive 

```cs
params.AddAuthor("lars");
```

givet et DocumentSearchParameter objekt med navnet params, hvilket selvfølgelig kan gøres med variable